src/RobotSimulator.elm

```diff
@@ -0,0 +1,305 @@
+module Main exposing (main, update, view)
+
+import Browser
+import Browser.Events
+import Html exposing (..)
+import Html.Attributes exposing (style)
+import Html.Events exposing (onClick)
+import Json.Decode as Decode
+import Time
+
+
+screenWidth =
+    500
+
+
+screenHeight =
+    500
```

Comment:
These shouldn't be hardcoded here. Just take them as additional parameters to the relevant functions.